### PR TITLE
MAID-3071: Add test for MissingGenesis

### PR DESCRIPTION
**Pull request by octol**
MissingGenesis malice is when a node that is part of the genesis group creates an event with no Observation::Genesis immediately after its initial event.

### CODE REVIEW

File: src/parsec.rs [hunk 1]
```diff
@@ -1798,6 +1798,72 @@ mod functional_tests {
...
+        let d_1 = Event::<Transaction, _>::new_from_observation(
+            d_0_hash,
+            Observation::Genesis(genesis),
+            &dave_contents.events,
+            &dave_contents.peer_list,
+        );
+        dave_contents.add_event(d_1);
+        let mut dave = Parsec::from_parsed_contents(dave_contents);
+
+        // Send sync event from Alice to Dave
```
Comment (madadam) [inline src/parsec.rs]: Minor, but we are not sending the sync event. The sync event is created by the recipient after processing all the events in the gossip message. So the comment should be something like "send gossip from Alice to Dave".

Comment (octol) [reply src/parsec.rs]: Ah yes good point! I'll update

File: src/parsec.rs [hunk 2]
```diff
@@ -1807,6 +1807,72 @@ mod functional_tests {
...
+                .filter_map(|payload| match payload {
+                    Observation::Accusation {
+                        ref offender,
+                        malice: Malice::MissingGenesis(hash),
+                    } => Some((offender, hash)),
+                    _ => None,
+                }).next()
+        );
+        assert_eq!(*offender, alice_id);
+        assert_eq!(*hash, a_1_hash);
```
Comment (maqi) [inline src/parsec.rs]: Do we also need to verify the events (a_0 and a_1) reveived via gossip have been added into receiver's graph?

Comment (octol) [reply src/parsec.rs]: Added in the latest commit, thanks :)

**Pull request merged**